test/test_sparse.py

```diff
@@ -3490,6 +3493,78 @@ def test(sparse_dims, nnz, with_size, new_size):
         test(4, 6, [7, 3, 1, 3, 1, 3], [7, 3, 1, 3, 2, 3])
         test(4, 6, [7, 3, 1, 3, 2, 1], [7, 3, 1, 3, 2, 3])
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy is not availible")
+    @onlyCPU
+    @dtypes(*all_types_and_complex_and(torch.bool))
+    def test_sparse_spdiags(self, device, dtype):
```

Comment:
The way this test is structured I think it's definitely worthwhile porting this over to OpInfo as a follow-up.

Reply:
I structured the tests so that change would be easier.